modules/apps/object/object-service/src/main/java/com/liferay/object/service/impl/ObjectDefinitionLocalServiceImpl.java

```diff
@@ -203,15 +205,19 @@ public ObjectDefinition addObjectDefinition(
 
 		objectDefinition.setActive(false);
 		objectDefinition.setLabel(externalReferenceCode);
-		objectDefinition.setModifiable(true);
+		objectDefinition.setModifiable(modifiable);
 		objectDefinition.setName(externalReferenceCode);
 		objectDefinition.setPluralLabel(externalReferenceCode);
 		objectDefinition.setScope(ObjectDefinitionConstants.SCOPE_COMPANY);
 		objectDefinition.setStorageType(
 			ObjectDefinitionConstants.STORAGE_TYPE_DEFAULT);
-		objectDefinition.setSystem(false);
+		objectDefinition.setSystem(system);
 		objectDefinition.setStatus(WorkflowConstants.STATUS_DRAFT);
 
+		if (objectDefinition.isUnmodifiableSystemObject() || !modifiable) {
```

Comment:
I don't know... I think that we need refactor all `isUnmodifiableSystemObject` to:

1. `isSystem()`
2. `isModifiableSystem()`
3. `isCustom()`

Because this, I didn' t create a new method... What do you think?

Reply:
okay, I'm just asking this because other add methods in Object*LocalServices are doing some validations in private methods.

I believe this case is similar to [ObjectRelationshipLocalServiceImpl.java#L840-L849](https://github.com/liferay/liferay-portal/blob/56e35080ae7108ff178c9f44dcd3a80f85178431/modules/apps/object/object-service/src/main/java/com/liferay/object/service/impl/ObjectRelationshipLocalServiceImpl.java#L840-L849), I was thinking about just moving this code into a private method. Do you think this looks good? :thinking: 

Please let me know what you think about it :+1: